tests/BrowscapTest/Coverage/ProcessorTest.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Copyright (c) 1998-2017 Browser Capabilities Project
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Refer to the LICENSE file distributed with this package.
+ *
+ * @category   BrowscapTest
+ * @copyright  1998-2017 Browser Capabilities Project
+ * @license    MIT
+ */
+
+namespace BrowscapTest\Coverage;
+
+use Browscap\Coverage\Processor;
+
+/**
+ * Class ExpanderTest
+ *
+ * @category   BrowscapTest
+ * @author     Jay Klehr <jay.klehr@gmail.com>
+ */
+class ProcessorTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @var \Browscap\Coverage\Processor
+     */
+    private $object = null;
+
+    /**
+     * @var string
+     */
+    private $resourceDir = __DIR__ . '/../../fixtures/coverage/';
+
+    /**
+     * Run before each test, creates a new Processor object
+     */
+    public function setUp()
+    {
+        $this->object = new Processor($this->resourceDir);
+    }
+
+    /**
+     * Data provider for the testJsonStructure test
+     */
+    public function jsonStructureProvider()
+    {
+        return [
+            ['test1.json', ['statementCount' => 5, 'branchCount' => 1, 'functionCount' => 1]],
+            ['test2.json', ['statementCount' => 15, 'branchCount' => 3, 'functionCount' => 3]],
+        ];
+    }
+
+    /**
+     * This test verifies that the different structures were extracted from the test JSON files
+     *
+     * @dataProvider jsonStructureProvider
+     */
+    public function testJsonStructure($fileName, $expected)
+    {
+        $coverage = $this->object->processFile($fileName, file_get_contents($this->resourceDir . $fileName), []);
+
+        self::assertSame($expected['statementCount'], count($coverage['statementMap']));
+        self::assertSame($expected['statementCount'], count($coverage['s']));
+
+        self::assertSame($expected['branchCount'], count($coverage['branchMap']));
+        self::assertSame($expected['branchCount'], count($coverage['b']));
+
+        self::assertSame($expected['functionCount'], count($coverage['fnMap']));
+        self::assertSame($expected['functionCount'], count($coverage['f']));
+    }
+
+    /**
+     * Data provider for the testCoverage test
+     */
+    public function coverageProvider()
+    {
+        return [
+            // No coverage
```

Comment:
Rather than commenting here, you can turn these into useful debugging info for phpunit by making them the array keys (then phpunit uses them to identify which iteration of the provider we're on with meaningful name)